src/Features/CSharp/Portable/Diagnostics/Analyzers/CSharpSimplifyTypeNamesDiagnosticAnalyzer.cs

```diff
@@ -118,53 +123,48 @@ private static bool IsCrefCandidate(SyntaxNode node)
                 return false;
             }
 
-            // For Crefs, currently only Qualified Crefs needs to be handled separately
-            if (node.Kind() == SyntaxKind.QualifiedCref)
+            if (node.ContainsDiagnostics)
             {
-                if (node.ContainsDiagnostics)
-                {
-                    return false;
-                }
+                return false;
+            }
 
-                var crefSyntax = (CrefSyntax)node;
-                if (!crefSyntax.TryReduceOrSimplifyExplicitName(model, out var replacementNode, out issueSpan, optionSet, cancellationToken))
-                {
+            SyntaxNode replacementSyntax;
+            if (node.IsKind(SyntaxKind.QualifiedCref, out QualifiedCrefSyntax crefSyntax))
+            {
+                if (!crefSyntax.TryReduceOrSimplifyExplicitName(model, out var replacement, out issueSpan, optionSet, cancellationToken))
                     return false;
```

Comment:
Missing braces (here and elsewhere)?

Reply:
So, my understanding is that this is now allowed :)